rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -1504,6 +1506,41 @@ SecRule ARGS "@rx \W{4}" \
     setvar:'tx.anomaly_score_pl3=+%{tx.warning_anomaly_score}'"
 
 
+#
+# This is a sibling of rule 942100 that adds checking of the last path segment.
+#
+# libinjection is more likely to fail when passing the full path. E.g. the following
+# string produces a match:
+# 999999.1 union select unhex(hex(version())) -- and 1=1
+# while this doesn't:
+# /999999.1 union select unhex(hex(version())) -- and 1=1\.
+# Therefore, we capture the last segment of the path and only match that with
+# libinjection. Incidentally, the last path segment is also the most likely
+# to be used for injection, other segments will most likely not be affected.
+#
+SecRule REQUEST_BASENAME "@detectSQLi" \
+    "id:942101,\
+    phase:2,\
+    block,\
+    capture,\
+    t:none,t:utf8toUnicode,t:urlDecodeUni,t:removeNulls,\
+    msg:'SQL Injection Attack Detected via libinjection',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    tag:'application-multi',\
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-sqli',\
+    tag:'OWASP_CRS/WEB_ATTACK/SQL_INJECTION',\
+    tag:'WASCTC/WASC-19',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'OWASP_AppSensor/CIE1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    ver:'OWASP_CRS/3.1.0',\
```

Comment:
Do we want 3.2.0 here?

Reply:
It's 3.1.0 across the board. I take it we are going to script-update it before 3.2.0-rc1.